llvm/lib/Target/AArch64/AArch64InstrInfo.td

```diff
@@ -381,6 +381,9 @@ def NoUseScalarIncVL : Predicate<"!Subtarget->useScalarIncVL()">;
 
 def UseSVEFPLD1R : Predicate<"!Subtarget->noSVEFPLD1R()">;
 
+def UseUnaryUndefPseudos
+    : Predicate<"!(Subtarget->hasSVE2() || (Subtarget->isStreaming() && Subtarget->hasSME2()))">;
```

Comment:
This looks like  `!HasSVE2orSME2` rather than the intended `!HasSVE2p2orSME2p2`?

Reply:
Fixed.